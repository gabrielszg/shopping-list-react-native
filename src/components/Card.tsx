import React from 'react';

import {View, Text, StyleSheet, Pressable} from 'react-native';

interface Props {
  modalIsOpen: boolean;
  setIsOpen: React.Dispatch<React.SetStateAction<boolean>>;
}

function Card({modalIsOpen, setIsOpen}: Props): JSX.Element {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>Sua lista de compras está vazia!</Text>

      <Text style={styles.body}>Adicione itens a sua lista de compras</Text>

      <Pressable
        style={styles.btnAddNewList}
        onPress={() => setIsOpen(!modalIsOpen)}>
        <Text style={styles.titleBtnAddNewList}>Começar minha lista</Text>
      </Pressable>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#fff',
    borderRadius: 10,
    marginTop: 160,
    marginLeft: 10,
    marginRight: 10,
    padding: 20,
  },

  title: {
    fontSize: 25,
    fontWeight: 'bold',
    marginVertical: 8,
    textAlign: 'center',
    color: '#000',
  },

  body: {
    fontSize: 15,
    color: '#000',
    textAlign: 'center',
    marginBottom: 10,
  },

  btnAddNewList: {
    alignSelf: 'center',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    width: 320,
    padding: 10,
    borderRadius: 5,
    border: 'none',
    backgroundColor: 'rgb(57, 57, 226)',
  },

  titleBtnAddNewList: {
    fontSize: 16,
    color: '#fff',
  },
});

export default Card;
