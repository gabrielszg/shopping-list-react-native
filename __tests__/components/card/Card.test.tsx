/**
 * @format
 */

import 'react-native';
import React from 'react';
import {describe, it, expect} from '@jest/globals';
import {render} from '@testing-library/react-native';
import Card from '../../../src/components/Card';

describe('Card Test', () => {
  it('title and description correctly displayed', () => {
    const {getByText} = render(
      <Card
        modalIsOpen={false}
        setIsOpen={function (): void {
          throw new Error('Function not implemented.');
        }}
      />,
    );

    const title = getByText('Sua lista de compras está vazia!');
    const description = getByText('Adicione itens a sua lista de compras');

    expect(title).toBeTruthy();
    expect(description).toBeTruthy();
  });

  it('button title correctly displayed', () => {
    const {getByText} = render(
      <Card
        modalIsOpen={false}
        setIsOpen={function (): void {
          throw new Error('Function not implemented.');
        }}
      />,
    );

    const btnTitle = getByText('Começar minha lista');
    expect(btnTitle).toBeTruthy();
  });
});
