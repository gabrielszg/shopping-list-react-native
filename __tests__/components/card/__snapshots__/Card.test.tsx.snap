// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "borderRadius": 10,
      "marginLeft": 15,
      "marginRight": 15,
      "marginTop": 135,
      "padding": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "#000",
        "fontSize": 25,
        "fontWeight": "bold",
        "marginVertical": 8,
        "textAlign": "center",
      }
    }
  >
    Sua lista de compras está vazia!
  </Text>
  <Text
    style={
      {
        "color": "#000",
        "fontSize": 15,
        "marginBottom": 10,
        "textAlign": "center",
      }
    }
  >
    Adicione itens a sua lista de compras
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgb(57, 57, 226)",
        "border": "none",
        "borderRadius": 20,
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 16,
        }
      }
    >
      Começar minha lista
    </Text>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "borderRadius": 10,
      "marginLeft": 15,
      "marginRight": 15,
      "marginTop": 135,
      "padding": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "#000",
        "fontSize": 25,
        "fontWeight": "bold",
        "marginVertical": 8,
        "textAlign": "center",
      }
    }
  >
    Sua lista de compras está vazia!
  </Text>
  <Text
    style={
      {
        "color": "#000",
        "fontSize": 15,
        "marginBottom": 10,
        "textAlign": "center",
      }
    }
  >
    Adicione itens a sua lista de compras
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgb(57, 57, 226)",
        "border": "none",
        "borderRadius": 20,
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 16,
        }
      }
    >
      Começar minha lista
    </Text>
  </View>
</View>
`;
